test/test_kernel.cpp

```diff
@@ -0,0 +1,267 @@
+
+        source.program("cuda_add").compile().configure(grid_block_no2, grid_block_no3).launch(args);
+
+        //B2C3. Comparing the results
+        //A. With hOut1
+        for (int i=0;i<5;i++) REQUIRE(hOut_1[i] == hostCompareOutput[i]);
+        
+        for (int j=5;j<10;j++){
+            REQUIRE(hOut_1[j] == counter);
+            counter++;
+        }
+
+        //B. With hOut2
+        counter = 1;
+
+        for (int i=0;i<5;i++) REQUIRE(hOut_2[i] == hostCompareOutput[i]);
+        
+        for (int j=5;j<10;j++){
+            REQUIRE(hOut_2[j] == counter);
+            counter++;
+        }
+
+        //C. With hOut3
+        counter = 1;
+
+        for (int i=0;i<5;i++) REQUIRE(hOut_3[i] == hostCompareOutput[i]);
+        
+        for (int j=5;j<10;j++){
+            REQUIRE(hOut_3[j] == counter);
+            counter++;
+        }
+
+        //D. With hOut4
+         counter = 1;
+
+        for (int i=0;i<5;i++) REQUIRE(hOut_4[i] == hostCompareOutput[i]);
+        
+        for (int j=5;j<10;j++){
+            REQUIRE(hOut_4[j] == counter);
+            counter++;
+        }
+    }
+
+    //B2D. Validating, that is impossible to have too few or many kernel-arguments in an array
+    SECTION("2D. Error results are produced, when the kernels are launched using too few or many"
+    " kernel arguments according to the kernel-functions"){
+        //B2D1. Declaration of all necessary kernel function inputs.    
+        Source source{"#include \"cuda_runtime.h\"\n"
+        "extern \"C\"\n"
+        "__global__ void cuda_add(int *x, int *y, int *out, int "
+        "datasize) {\n"
+        " int i = threadIdx.x;\n"
+        " out[i] = x[i] + y[i];\n"
+        "}", headers};
+
+        dim3 grid_test(1), block_test(10);
+        const CUresult error{CUDA_ERROR_LAUNCH_OUT_OF_RESOURCES};
+        
+        args.clear();
+        args.emplace_back(KernelArg{hX, bufferSize});
+        args.emplace_back(KernelArg{hY, bufferSize});
+        args.emplace_back(KernelArg{hOut_1, bufferSize, true});
+        args.emplace_back(KernelArg(&datasize));
+        args.emplace_back(KernelArg{hOut_2, bufferSize, true});
+        args.emplace_back(KernelArg{hOut_3, bufferSize, true});
+        args.emplace_back(KernelArg{hOut_4, bufferSize, true});
+
+        REQUIRE_NOTHROW(source.program("cuda_add").compile().configure(grid_test, block_test).launch(args));
+        
+        //B2D2. Comparing the results
+        for (int i=0;i<10;i++) REQUIRE(hOut_1[i] == hostCompareOutput[i]);
+
+        //B2B3. Validating, that it is impossible to launch a kernel using too few kernel arguments
+        args.clear();
+        args.emplace_back(KernelArg{hX, bufferSize});
+        args.emplace_back(KernelArg{hY, bufferSize});
+        args.emplace_back(KernelArg{hOut_1, bufferSize, true});
+
+        signal(SIGSEGV,segmentfunction);
+
+        source.program("cuda_add").compile().configure(grid_test, block_test).launch(args);
```

Comment:
Probleme mit dem Abfangen von Segmentation fault. (SIGSEVS)

Reply:
yeah apparently this is non trivial to prevent so we just have to hope the user will do it correctly.
we're not going to test for this